### DISCUSSION TITLE
Only fetch specific attributes when getting mails from folder

### DISCUSSION THREAD

**Discussion by threeFFF** [O365/python-o365#857] [Q&A] [answered]
Hello,

I want to monitor the inbox folder of a mailbox with up to 2500 messages in that folder. The loop

```
for message in mail_folder.get_messages(limit=None):
    [...]
```

takes about 50 seconds to finish. The point is that I am not interested in the e-mails themselves, I just want to know the received time to check if some e-mails have been too long in the folder.

When using the exchangelib module in place of the O365 module, there is the only() method to not fetch everything but just the attributes I need:

```
for email in folder.all().only('subject', 'datetime_received'):
    [...]
```

The use of only() significantly decreases the time needed to get the messages from the folder. Unfortunately, I could not find anything similar in O365. Or did I miss something?

Best regards,
Michael

**Comment by alejcas** [comment, marked as answer]
Use select!

**Reply by threeFFF** [reply]
Thank you very much, that was the solution! Reading the mailbox is now much faster.
